lib/ember-cli/app.rb

```diff
@@ -84,45 +82,34 @@ def wait
       MSG
     end
 
-    def ember_path
-      @ember_path ||= app_path.join("node_modules", ".bin", "ember").tap do |path|
-        fail <<-MSG.strip_heredoc unless path.executable?
-          No local ember executable found. You should run `npm install`
-          inside the #{name} app located at #{app_path}
-        MSG
+    def method_missing(method_name, *)
+      if path_method = supported_path_method(method_name)
+        paths.public_send(path_method)
+      else
+        super
```

Comment:
@seanpdoyle well, it'll nehave differently. Now methods like `dist_path` delegate to `dist` method of the `paths` object. I like this `_path` suffix because it reveals the intention. I agree that all these method missing stuff could be a bit of an over engineering, but I love the result.

Far less verbose way would be something like this:

```ruby
delegate :dist, :assets, :tmp, lockfile, to: :paths, suffix: :path
```

but then you'll hav to manually mention every delegated method, and update it every time you need to introduce another one. Feels messy.